### PR TITLE
feat(ST-140): delete assignment endpoint

### PR DESCRIPTION
**Pull request by ghost**
## Description of Change

- Updated assignments.controller for implementing deleting exam
- Updated assignments.service for implementing the logic
- Update assignments.repository

## Associated Ticket Link(s)
- https://trello.com/c/n6iiILmB

## Screenshots / Screen Recordings
N/A

### CODE REVIEW

File: src/assignments/assignments.service.ts [hunk 1]
```diff
@@ -64,4 +64,14 @@ export class AssignmentsService {
 
     return this.assignmentsRepository.updateOne(assignment, updateAssignmentDto);
   }
+
+  async deleteAssignment(assignmentId: number): Promise<void> {
+    const assignment = await this.assignmentsRepository.findOneOrFail(assignmentId);
+
+    const fileKey = assignment.fileUrl.split("project-dev-bucket/")[1];
+
+    await this.fileUploadsService.deleteFromS3(fileKey);
```
Comment (ishmam-iqbal-sazim) [inline src/assignments/assignments.service.ts:79]: Probably best to handle fail case before updating database

**Review by ragib-hossain-sazim — APPROVED** [review]

File: src/assignments/assignments.controller.ts [hunk 2]
```diff
@@ -37,18 +40,29 @@ export class AssignmentsController {
...
   @Get(":classroomId/assignments")
-  getAssignments(@Param("classroomId") classroomId: number) {
-    return this.assignmentsService.getAssignmentsByClassroomId(classroomId);
+  async getAssignments(@Param("classroomId") classroomId: number): Promise<Assignment[]> {
+    const assignments = await this.assignmentsService.getAssignmentsByClassroomId(classroomId);
+    return this.assignmentsSerializer.serializeMany(assignments);
   }
 
-  @Put(":classroomId/assignments/:assignmentId")
+  @Put("assignments/:assignmentId")
```
Comment (ishmam-iqbal-sazim) [inline src/assignments/assignments.controller.ts:48]:
```suggestion
  @Put(":classroomId/assignments/:assignmentId")
```

File: src/assignments/assignments.controller.ts [hunk 3]
```diff
@@ -37,18 +40,29 @@ export class AssignmentsController {
...
+  ): Promise<Assignment> {
+    const assignment = await this.assignmentsService.updateOne(
+      assignmentId,
+      updateAssignmentDto,
+      file,
+    );
+    return this.assignmentsSerializer.serialize(assignment);
+  }
+
+  @Delete("assignments/:assignmentId")
```
Comment (ishmam-iqbal-sazim) [inline src/assignments/assignments.controller.ts:63]:
```suggestion
  @Delete(":classroomId/assignments/:assignmentId")
```

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comments

**Pull request merged**